interfaces/matlab_experimental/1D/Domain1D.m

```diff
@@ -0,0 +1,696 @@
+classdef Domain1D < handle
+    % Domain1D Class ::
+    %
+    %     >> d = Domain1D(a, b)
+    %
+    % :param a:
+    %    String type of domain. Possible values are:
```

Comment:
The constructor should be called by the specialized class, and use the new factories introduced in #1445 (they use different names)